newbank/server/Commands/BorrowCommand.java

```diff
@@ -0,0 +1,156 @@
+      return;
+    }
+
+    Offer offer = MicroLoanMarketPlace.getInstance().getOffers().get(offerNumber);
+    borrowingLength = offer.getBorrowingLengthInMonth();
+
+    if (input.length == 2) {
+      if (!validInt(input[1])) {
+        response.invalidRequest(
+            "Please enter correct format for amounts. Please see help and try again.");
+        return;
+      }
+      borrowingLength = Integer.parseInt(input[1]);
+    }
+
+    if (borrowingLength > offer.getBorrowingLengthInMonth()) {
+      response.failed("Length selected is longer than max allowed. Please try again.");
+      return;
+    }
+
+    Customer customer = request.getCustomer();
+    Account creditedAccount = findCreditingAccount(customer, response);
+
+    if (creditedAccount == null) {
+      response.failed("No account available to credit. Please try again.");
+      return;
+    }
+    RepaymentCalculator rc = new RepaymentCalculator();
+    BigDecimal repaymentAmount = rc.calculateRepayments(offer.getAmount(), offer.getInterestRate(), borrowingLength);
+
+    String confirmationMessage =
+        "Please confirm your loan:"
+            + System.lineSeparator()
+            + "Loan Amount: "
+            + offer.getAmount().toPlainString()
+            + "GBP"
+            + System.lineSeparator()
+            + "Interest rate: "
+            + offer.getInterestRate().multiply(BigDecimal.valueOf(100)).toPlainString()
+            + "%"
+            + System.lineSeparator()
+            + "Borrowing length: "
+            + borrowingLength
+            + System.lineSeparator()
+            + "Your monthly repayment will be: "
+            + repaymentAmount
+                .toPlainString()
+            + System.lineSeparator()
+            + "Do you wish to proceed?";
+
+    if (!response.confirm(confirmationMessage)) {
+      response.failed("Action cancelled.");
+      return;
+    }
+
+    Loan loan =
+        new Loan(offer, offer.getAmount(), BigDecimal.valueOf(0), creditedAccount, borrowingLength);
+
+    customer.addLoan(loan);
+
+    creditedAccount.moneyIn(offer.getAmount());
+    MicroLoanMarketPlace.getInstance().removeOffer(offerNumber);
+
+    response.succeeded(
+        "You have successfully borrowed " + offer.getAmount().toPlainString() + "GBP.");
+  }
+
+  private static Account findCreditingAccount(Customer customer, NewBankCommandResponse response) {
+    List<Account> accounts =
+        customer.enumAccountNumbers().stream()
+            .map(customer::getAccountFromNumber)
+            .filter(account -> !account.getAccountType().equals(Account.AccountType.LENDING))
+            .collect(Collectors.toList());
+    return accounts.size() == 0
+        ? null
+        : accounts.size() == 1 ? accounts.get(0) : queryAccount(customer, response, accounts);
+  }
+
+  private static Account queryAccount(
+      Customer customer, NewBankCommandResponse response, List accountList) {
```

Comment:
List should take the parameter <Account> here.

Reply:
Hi, just changed, also I made sure the crediting account was GBP